CIP-0165/README.md

```diff
@@ -0,0 +1,417 @@
+- `slot_no` : `u64` identifier of the blockchain point (hash/slot).
+
+**Policy:**
+
+- appears once in the file header;
+- must be read and verified first;
+- supports magic bytes for file recognition.
+
+#### CHUNK Records
+
+**Purpose:** group entries for streaming and integrity; maintain global canonical order, see namespace and entries for more details.
+
+**Structure:**
+
+- `chunk_seq` : `u64` — sequence number of the record
+- `chunk_format` : `u8` - format of the chunks
+- `namespace` : `bstr` — namespace of the values stored in the CHUNK
+- `entries` — CBOR encoded list of unlimited length
+- `footer {entries_count: u64, chunk_hash: u64}` — hash value of the chunk of data, is used to keep integrity of the file.
+
+**Policy:**
+
+- chunk size \~8–16MiB; footer required;
+- data is stored in deterministically defined global order; In the lexical order of the keys;
+- all keys in the record must be unique;
+- all key-values in the record must refer to the same namespace;
+- readers should verify footer before relying on the data;
+- `chunk_hash = H(concat [ digest(e) | e in entries ])`;
+- all keys in `CHUNK` `n` must be lexicographically lower than all keys in `CHUNK` `n+1`.
+
+The format proposes support of data compression. For future-compatibility the format is described by the `chunk_format` field, and following variants are introduced:
+
+| Code | Name  | Description                                   |
+| ---- | ----- | --------------------------------------------- |
+| 0x00 | RAW   | Raw CBOR Entries                              |
+| 0x01 | ZSTD  | All entries are compressed with seekable zstd |
+| 0x02 | ZSTDE | Compress each value independently             |
+
+When calculating and verifying hashes, it's build over the uncompressed data.
+
+#### MANIFEST Record
+
+**Purpose:** index of chunks and information for file integrity check.
+
+**Structure:**
+
+- `total_entries`: `u64` — number of data entries in the file (integrity purpose only)
+- `total_chunks`: `u64` — number of chunks in the file (integrity purpose only)
+- `root_hash`: **Merkle root** of all `entry_e` in the chosen order, see verification for details
+- `namespace_hashes`: CBOR table of Merkle roots for each namespace, mapping namespace name to the hash in a multihash format
+- `prev_manifest_offset`: `u64` — offset of the previous manifest (used with delta files), zero if there is no previous manifest entry
+- `summary`: `{ created_at, tool, comment? }`
+
+**Policy:** used to verify all the chunks.
+
+#### DELTA Record
+
+```text
+TODO: exact contents of the DELTA record will be defined later, currently we describe
+a high-level proposal.
+```
+
+**Purpose:** Delta records are used to build iterative updates, when base format is created and we want to store additional transactions in a fast way. Delta records are designed to be compatible with UTxO-HD, LSM-Tree or other storage types where it's possible to stream list of updates.
```

Comment:
Which parts of the ledger state will record deltas and how? What's their format?

Most importantly why do we need to record deltas, why not resolve those deltas and just store the final data?